web/Controllers/PropertyController.cs

```diff
@@ -140,6 +140,24 @@ public ActionResult AddNewProperty_Cancel()
         // Go to View Properties page
         return RedirectToAction("ViewProperties", "Landlord");
     }
+    
+    [Authorize(Roles = "Landlord")]
+    public ActionResult DeleteProperty(int propertyId)
+    {
+        PropertyDbModel? propDB = _propertyService.GetPropertyByPropertyId(propertyId);
+        if (propDB == null)
+        {
+            _logger.LogWarning("Property with ID {PropertyId} does not exist", propertyId);
+            return RedirectToAction("Error", "Home", new { status = 404 });

```

Comment:
This should do almost the same thing (it will not redirect, instead just returning a 404 straight away, but that's probably better)
```suggestion
            return StatusCode(404);
```